ra2ce/graph/vector_network_wrapper.py

```diff
@@ -0,0 +1,321 @@
+
+        # to graph
+        digraph = nx.DiGraph(crs=gdf.crs, approach="primal")
+        for index, row in gdf.iterrows():
+            from_node = row.geometry.coords[0]
+            to_node = row.geometry.coords[-1]
+            digraph.add_node(from_node, geometry=Point(from_node))
+            digraph.add_node(to_node, geometry=Point(to_node))
+            digraph.add_edge(
+                from_node,
+                to_node,
+                geometry=row.pop(
+                    "geometry"
+                ),  # **row TODO: check if we do need all columns
+            )
+
+        if is_directed:
+            return digraph
+        return digraph.to_undirected()
+
+    @staticmethod
+    def setup_network_edges_and_nodes_from_graph(
+        graph: nx.Graph,
+    ) -> gpd.GeoDataFrame:
+        """Sets up network nodes and edges from a given graph.
+
+        Args:
+            graph (nx.Graph): Input graph with geometry for nodes and edges.
+                Must contain "crs" as graph property.
+
+        Returns:
+            gpd.GeoDataFrame: GeoDataFrame representing the network edges with "edge_fid", "node_A", and "node_B".
+            gpd.GeoDataFrame: GeoDataFrame representing the network nodes with "node_fid".
+        """
+
+        # TODO ths function use conventions. Good to make consistant convention with osm
+        nodes, edges = momepy.nx_to_gdf(graph, nodeID="node_fid")
+        edges["edge_fid"] = (
+            edges["node_start"].astype(str) + "_" + edges["node_end"].astype(str)
+        )
+        edges.rename(
+            {"node_start": "node_A", "node_end": "node_B"}, axis=1, inplace=True
+        )
+        if not nodes.crs:
+            nodes.crs = graph.graph["crs"]
+        if not edges.crs:
+            edges.crs = graph.graph["crs"]
+        return edges, nodes
+
+    def setup_network_from_vector(
+        self,
+    ) -> Tuple[nx.MultiGraph, gpd.GeoDataFrame]:
+        """Sets up a network from vector files.
+
+        Returns:
+            nx.MultiGraph: MultiGraph representing the graph.
+            gpd.GeoDataFrame: GeoDataFrame representing the network.
+        """
+        files = self.network_dict["files"]
+        file_crs = self.network_dict["file_crs"]
+        is_directed = self.network_dict["is_directed"]
+
+        gdf = self._read_vector_to_project_region_and_crs(
+            vector_filenames=files, crs=file_crs
+        )
+        gdf = VectorNetworkWrapper.clean_vector(gdf)
+        graph = VectorNetworkWrapper.setup_graph_from_vector(
+            gdf, is_directed=is_directed
+        )
+        edges, nodes = VectorNetworkWrapper.setup_network_edges_and_nodes_from_graph(
+            graph
+        )
+        graph_complex = nut.graph_from_gdf(edges, nodes, node_id="node_fid")
+        return graph_complex, edges
+
+    def _read_vector_to_project_region_and_crs(
+        self, vector_filenames: List[Path], crs: Union[int, str]
```

Comment:
You don't need the `typing.List`, you can directly use `list`.
```suggestion
        self, vector_filenames: list[Path], crs: Union[int, str]
```